modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-volley/api.mustache

```diff
@@ -0,0 +1,129 @@
+    private val context: Context,
+    private val requestQueue: Lazy<RequestQueue> = lazy(initializer = {
+        Volley.newRequestQueue(context.applicationContext)
+    }),
+    private val requestFactory: IRequestFactory = RequestFactory(),
+    private val basePath: String = "{{{basePath}}}",
+    private val postProcessors :List <(Request<*>) -> Unit> = listOf()) {
+
+    {{#operation}}
+    /**
+    * {{summary}}
+    * {{notes}}
+    {{#allParams}}   * @param {{paramName}} {{description}}
+    {{/allParams}}   * @return {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}
+    */
+    {{#isDeprecated}}
+    @Deprecated("This api was deprecated")
+    {{/isDeprecated}}
+    suspend fun {{operationId}}({{^allParams}}){{/allParams}}{{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}}{{^-last}}, {{/-last}}{{#-last}}{{#operationAuthMethod}}, opAuthHeaderFactory = () -> map<String, String>{{/operationAuthMethod}}){{/-last}}{{/allParams}}: {{#returnType}}{{{returnType}}}?{{/returnType}}{{^returnType}}Unit{{/returnType}} {
+        {{#bodyParam}}
+        val body: Any? = {{paramName}}
+        {{/bodyParam}}
+        {{^bodyParam}}
+        val body: Any? = null
+        {{/bodyParam}}
+        {{#allParams}}
+        {{#required}}
+        // verify the required parameter '{{paramName}}' is set
+        // This is probably taken care of by non-null types anyway
+        requireNotNull({{paramName}})
+        {{/required}}
+        {{/allParams}}
+
+        val contentTypes : Array<String> = arrayOf({{#consumes}}"{{{mediaType}}}"{{^-last}},{{/-last}}{{/consumes}})
+        val contentType: String = if (contentTypes.isNotEmpty()) { contentTypes.first() } else { "application/json" }
+
+        // Do some work or avoid some work based on what we know about the model, 
+        // before we delegate to a pluggable request factory template
+        // The request factory template contains only pure code and no templates
+        // to make it easy to override with your own.
+
+        // create path and map variables
+        val path = "{{{path}}}"{{#pathParams}}.replace("{" + "{{baseName}}" + "}", IRequestFactory.escapeString({{{paramName}}}.toString())){{/pathParams}};
+
+        // form params
+        val formParams = mapOf<String, String>(
+            {{#formParams}}
+                    "{{baseName}}" to IRequestFactory.parameterToString({{paramName}}),
+            {{/formParams}}
+        )
+
+
+        // TODO: Cater for allowing empty values
+        // TODO, if its apikey auth, then add the header names here and the hardcoded auth key
+        // Only support hard coded apikey in query param auth for when we do this first path
+        val queryParams = mapOf<String, String>(
+            {{#queryParams}}
+            "{{baseName}}" to IRequestFactory.parameterToString({{paramName}}),
+            {{/queryParams}}
+        ).filter { it.value != ""}
```

Comment:
@alisters could you please replace `it.value != ""` with `it.value.isNotEmpty()`?

Reply:
Done